packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -87,6 +87,16 @@ export interface CommonAutoScalingGroupProps {
    */
   readonly keyName?: string;
 
+  /**
+   * The SSH keypair to grant access to the instance.
+   *
+   * `keyName`, `launchTemplate` and `mixedInstancesPolicy` must not be specified
```

Comment:
We're missing [this](https://github.com/aws/aws-cdk/blob/bf2cf51bf1e8458097b23e7f580ad0978b7e4cca/packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts#L1719) validation for `keyPair`.

Reply:
Fixed in https://github.com/aws/aws-cdk/pull/29679/commits/73e2e99e7f3edd757ea4123826a877e04d69da56